app/src/main/java/com/sdp13epfl2021/projmag/Form.kt

```diff
@@ -1,76 +1,165 @@
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK && requestCode == REQUEST_VIDEO_ACCESS) {
+            if (data?.data != null) {
+                // THIS IS THE VID URI
+                videoUri = data.data
+
+                val playVidButton = findViewById<Button>(R.id.play_video)
+                val vidView = findViewById<VideoView>(R.id.videoView)
+                val mediaController = MediaController(this)
+
+                FormHelper.playVideoFromLocalPath(
+                    playVidButton,
+                    vidView,
+                    mediaController,
+                    videoUri!!
+                )
+            }
+        }
+    }
+
+    /**
+     * Extract string text content form an EditText view
+     */
+    private fun getTextFromEditText(id: Int): String = findViewById<EditText>(id).run {
+        text.toString()
+    }
+
+    /**
+     * Show a toast message on the UI thread
+     * This is useful when using async callbacks
+     */
+    private fun showToast(msg: String) = runOnUiThread {
+        Toast.makeText(this, msg, Toast.LENGTH_LONG).show()
+    }
+
+    /**
+     * Construct a Project with data present in the view
+     */
+    private fun constructProject(): Result<ImmutableProject> {
+        return ImmutableProject.build(
+            id = "",
+            name = getTextFromEditText(R.id.form_edit_text_project_name),
+            lab = getTextFromEditText(R.id.form_edit_text_laboratory),
+            teacher = getTextFromEditText(R.id.form_edit_text_teacher),
+            TA = getTextFromEditText(R.id.form_edit_text_project_TA),
+            nbParticipant = try {
+                getTextFromEditText(R.id.form_nb_of_participant).toInt()
+            } catch (_: NumberFormatException) {
+                0
+            },
+            masterProject = findViewById<CheckBox>(R.id.form_check_box_MP).isChecked,
+            bachelorProject = findViewById<CheckBox>(R.id.form_check_box_SP).isChecked,
+            isTaken = false,
+            description = getTextFromEditText(R.id.form_project_description),
+            assigned = listOf(),
+            tags = listOf("Default-tag")
+        )
+    }
+
+
+    /**
+     * Finish the activity from another thread
+     * Useful when using async callbacks
+     */
+    private fun finishFromOtherThread() = runOnUiThread {
+        finish()
+    }
+
+    /**
+     * Submit project and video with information in the view.
+     * Expected to be called when clicking on a submission button on the view
+     */
+    private fun submit(view: View) = Firebase.auth.let { auth ->
+        setSubmitButtonEnabled(false) // disable submit, as there is a long time uploading video
+        ProjectUploader(
+            Utils(this).projectsDatabase,
+            FirebaseFileDatabase(
+                FirebaseStorage.getInstance(),
+                auth
+            ),
```

Comment:
There is a filedatabse on Utils
```suggestion
        val utils = Utils(this)
        ProjectUploader(
            utils.projectsDatabase,
            utils.fileDatabase,
```